### PR TITLE
fix: remove distutils leftovers

### PR DESCRIPTION
**Pull request by drupol**
Context: https://github.com/tensorflow/tensorflow/issues/58073
Applied patch: https://gitlab.archlinux.org/archlinux/packaging/packages/tensorflow/-/blob/fff04b07ca2bc6758f81505cc396c91f7466c08a/tensorflow-2.16.1-python-distutils-removal.patch

Credits to @loqs

### CODE REVIEW

File: tensorflow/lite/python/convert.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 # ==============================================================================
 """Converts a frozen graph into a TFLite FlatBuffer."""
 
-import distutils.spawn
+import shutil
```
Comment (GaetanLepage) [inline tensorflow/lite/python/convert.py:17]: Maybe you should preserve alphabetical ordering here.

File: tensorflow/api_template.__init__.py [hunk 2]
```diff
@@ -27,7 +27,10 @@
 """
 # pylint: disable=g-bad-import-order,protected-access,g-import-not-at-top
 
-import distutils as _distutils
+try:
+  import distutils as _distutils
+except ImportError:
+  pass
```
Comment (mihaimaruseac) [inline tensorflow/api_template.__init__.py:33]: Let's completely remove `distutils` here instead of using `try-except` and silence the failure

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/api_template.__init__.py [hunk 3]
```diff
@@ -100,8 +99,11 @@
 if "getsitepackages" in dir(_site):
   _site_packages_dirs += _site.getsitepackages()
 
-if "sysconfig" in dir(_distutils):
-  _site_packages_dirs += [_distutils.sysconfig.get_python_lib()]
+try:
+  if "sysconfig" in dir(_distutils):
+    _site_packages_dirs += [_distutils.sysconfig.get_python_lib()]
+except NameError:
+  pass
```
Comment (drupol) [inline tensorflow/api_template.__init__.py:106]: Do you have pointers on how to remove this part?

Comment (mihaimaruseac) [reply tensorflow/api_template.__init__.py:106]: I think it should be ok to fully remove. If testing shows that this is a problem, then let's revisit.

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
The lint error is not from this change, so it should not be blocking.

**Comment by mihaimaruseac** [comment]
If I read the status correctly, there's nothing blocking. I cannot also approve internally, so we need to wait until ~tomorrow. After the internal approval there will be a few more CI jobs before merging. Sorry for the delay and thank you for the PR

**Pull request merged**